modules/genomic_browser/php/provisioners/cnvprovisioner.class.inc

```diff
@@ -0,0 +1,111 @@
+<?php declare(strict_types=1);
+/**
+ * PHP version 7
+ *
+ * @category Genomic_Browser
+ * @package  Main
+ * @author   Xavier Lecours <xavier.lecours@mcin.ca>
+ *           Alizée Wickenheiser <alizee.wickenheiser@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://github.com/aces/Loris-Trunk
+ */
+
+namespace LORIS\genomic_browser\provisioners;
+
+/**
+ * The CnvProvisioner class.
+ *
+ * PHP version 7
+ *
+ * @category Genomic_Browser
+ * @package  Main
+ * @author   Xavier Lecours <xavier.lecours@mcin.ca>
+ *           Alizée Wickenheiser <alizee.wickenheiser@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://github.com/aces/Loris-Trunk
+ */
+class CnvProvisioner extends \LORIS\Data\Provisioners\DBObjectProvisioner
+{
+    private $_and = '';
+
+    /**
+     * Create a RowProvisioner
+     */
+    function __construct()
+    {
+        $user = \User::singleton();
+        if (!$user->hasPermission('genomic_browser_view_allsites')) {
```

Comment:
I don't understand why this is done in SQL with one permission, but there's another (unused?) function below that does it with UserSiteMatch and a different permission.

Reply:
Hi @driusan I removed the UserSiteMatch functions since they weren't being used. The query here is taken from the existing logic of the module. I'm unsure how I should change it if still needed because the query is slightly altered if user has  `genomic_browser_view_allsites` permission.